analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/SymbolicExecution/SymbolicValueAssertions.cs

```diff
@@ -0,0 +1,164 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2022 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using FluentAssertions.Common;
+using FluentAssertions.Execution;
+using FluentAssertions.Primitives;
+using SonarAnalyzer.SymbolicExecution;
+using SonarAnalyzer.SymbolicExecution.Roslyn;
+
+namespace SonarAnalyzer.UnitTest.TestFramework.SymbolicExecution
+{
+    public static class SymbolicValueExtensions
+    {
+        public static SymbolicValueAssertions Should(this SymbolicValue instance) =>
+            new(instance);
+    }
+
+    public class SymbolicValueAssertions : ReferenceTypeAssertions<SymbolicValue, SymbolicValueAssertions>
+    {
+        protected override string Identifier => "symbolicValue";
+
+        public SymbolicValueAssertions(SymbolicValue subject) : base(subject) { }
+
+        public AndWhichConstraint<SymbolicValueAssertions, SymbolicConstraint> HaveConstraint(SymbolicConstraint constraint, string because = "", params object[] becauseArgs)
+        {
+            Execute.Assertion
+                .BecauseOf(because, becauseArgs)
+                .ForCondition(Subject != null)
+                .FailWith("The {context:symbolicValue} is not present and can not have constraint {0}.", constraint)
```

Comment:
Clearer message. Applies everywhere below too
```suggestion
                .FailWith("The {context:symbolicValue} is null and can not have constraint {0}.", constraint)
```